### PR TITLE
Switch to templates for interface edit.

### PR DESCRIPTION
**Pull request by justinlittman**
Rather than create the form inputs programmatically in JS, this switches to templates.

This is in preparation for work on #615, as it will make enhancing the input form easier.

### CODE REVIEW

File: libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html [hunk 1]
```diff
@@ -99,29 +99,54 @@ <h3>MARC Processing Rules</h3>
...
   <section id="remove-fields">
-
-  <div class="form-group" style="margin-top: 40px;">
-    <div class="col-sm-2">
-      <strong>Remove Fields</strong>
-    </div>
-    <div class="col-sm-1">
-      <button class="btn btn-primary col-sm-offset-1"
-        onclick="newRemoveFieldRule(); return false;">Add Rule</button>
+    <template id="remove-fields-template">
```
Comment (justinlittman) [inline libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html:102]: Here's the template for removing fields.

File: libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html [hunk 2]
```diff
@@ -99,29 +99,54 @@ <h3>MARC Processing Rules</h3>
...
-  </div>
-
   </section>
 
   <section id="move-fields">
-
-  <div class="form-group" style="margin-top: 40px">
-    <div class="col-sm-2">
-      <strong>Move Fields</strong>
+    <template id="move-fields-template">
```
Comment (justinlittman) [inline libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html:126]: Here's the template for moving fields.

File: libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html [hunk 3]
```diff
@@ -168,50 +193,32 @@ <h3>MARC Processing Rules</h3>
    * Creates a form element for a new Remove Field Rule and adds it to the form.
    */
   function newRemoveFieldRule(marcField='') {
-    const removeFields = document.getElementById("remove-fields");
-    const id = getId("remove-field");
-
-    const formGroup = element("div", {"class": "form-group remove-field"});
```
Comment (justinlittman) [inline libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html:174]: No longer building the form in JS.

File: libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html [hunk 4]
```diff
@@ -168,50 +193,32 @@ <h3>MARC Processing Rules</h3>
...
+    const removeFields = document.getElementById("remove-fields");
+    removeFields.appendChild(clone);
   }
 
   /**
    * Creates a Move Field Rule element and adds it to the form.
    */
-  function newMoveFieldRule(fromField='', toField='') {
-    const moveFields = document.getElementById("move-fields");
-    const formGroup = element("div", {"class": "form-group move-field"});
```
Comment (justinlittman) [inline libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html:191]: Same here.

File: libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html [hunk 5]
```diff
@@ -222,9 +229,26 @@ <h3>MARC Processing Rules</h3>
     form.removeChild(field);
   }
 
-  function getId(name) {
-    const epoch = new Date().getTime();
-    return `${name}-${epoch}`;
+  function updateInputTemplate(clone, id, now, value) {
```
Comment (justinlittman) [inline libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html:232]: These next few methods are for updating the cloned template.

**Review by aaron-collier — APPROVED** [review]

File: libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html (hunk 3 above)
Comment (edsu) [reply libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html:174]: This is a huge improvement, thanks for taking the time to do this @justinlittman. I had looked at using a template initially but wasn't sure if I could do it properly in JavaScript without a library. I will definitely remember that it's possible!

Comment (justinlittman) [reply libsys_airflow/plugins/vendor_app/templates/vendors/interface-edit.html:174]: I picked this up from @jcoyne, so credit is his.

**Pull request merged**